sstables/sstables.cc

```diff
@@ -1504,7 +1504,9 @@ future<> seal_summary(summary& s,
     s.header.size = s.entries.size();
     s.header.size_at_full_sampling = sstable::get_size_at_full_sampling(state.partition_count, s.header.min_index_interval);
 
-    assert(first_key); // assume non-empty sstable
+    if (!first_key) {
+        throw malformed_sstable_exception(format("Cannot seal summary because the SSTable has no keys written to it"));
+    }
```

Comment:
malformed_sstable is not accurate when we're writing the sstable.

on_internal_error() is better.

Also better to provide extra context.

Reply:
The intention is to not bring the node down, but if we're trying to seal an empty sstable (e.g. in the context of compaction, which can survive failures), then something got really bad in the higher layers. What about a runtime_error exception? The positive side of bringing the node is that it immediately draws operator attention for the issue.

Reply:
there are only a few cases that come to mind that malfored_sstable might make sense on write path, which is when writing out-of-order tokens. malformed sstable detected on the fly. or with integrity checker enabled, where the data written is not what we read back afterwards (due to corruption caused by disk / fs).

Reply:
on_internal_error() doesn't bring the node down.

Reply:
> on_internal_error(

Oops, I incorrectly thought it invoked abort().

Reply:
Ah that behavior is conditional

```
/// Controls whether on_internal_error() aborts or throws. The default
/// is to throw.
/// \returns the current abort_on_internal_error state.
bool set_abort_on_internal_error(bool do_abort) noexcept;
```